### ISSUE TITLE
Is it possible to perform transfer learning by further training the model with a custom dataset?

### ISSUE THREAD

**Issue by monocongo** [NVIDIA/DeepLearningExamples#218]
I want to train this model (PyTorch SSD/ResNet50) for detection of several object classes not found in the COCO dataset.

The README describes how to train this model from scratch using COCO as the input dataset. I would like to instead take some pre-computed weights from the training on COCO and use these as a starting point when performing further training of the model (transfer learning) using my own custom dataset of images/annotations as the training input. This process isn't described here as far as I can see, but maybe it's not too much different from the process of training the model from scratch with COCO as the input dataset?

If anyone can advise I will very much appreciate any suggestions. Thanks in advance!

**Comment by monocongo** [comment]
Also, are there weights corresponding to the training of this model using COCO as the input? It'd be nice to not have to run that initial training if I can just download the pre-trained weights instead.

**Comment by mwawrzos** [comment]
We are starting to publish pre-trained weights in https://ngc.nvidia.com/catalog/models. You can use that to initialize your network. If you want to play with the model interactively, I encourage the PyTorch Hub: https://pytorch.org/hub/nvidia_deeplearningexamples_ssd/

To solve your first problem, that is training a pre-trained model on a similar, but still a different task you can search for **fine-tuning**. We don't have en example published yet. However, if you will have a working example, don't hesitate to create a Merge Request. Collaboration is more than welcome.

**Comment by kobrica** [comment]
You can perform fine-tuning by freezing all of the layers of ResNet, except last few (or only last conv layer), which is done by setting their learning rate to 0, stoping backpropagation and then training as usual.
